Index: debian/control.in
===================================================================
--- debian/control.in	(revision 10906)
+++ debian/control.in	(working copy)
@@ -1,5 +1,5 @@
 Source: amsn
-Section: x11
+Section: user/x11
 Priority: optional
 Maintainer: Theodore Karkoulis <bilbo@debian.org>
 Build-Depends: debhelper (>> 4.0.0), tcl#TCL_VERSION#-dev, tk#TK_VERSION#-dev, libpng12-dev, libjpeg62-dev
Index: configure.ac
===================================================================
--- configure.ac	(revision 10906)
+++ configure.ac	(working copy)
@@ -357,7 +357,7 @@
 dnl do some OS specific stuff here
 
 dnl Check for libstdc++
-AC_CHECK_LIB(stdc++,main,CXX_LIB="-lstdc++", AC_MSG_ERROR(stdc++ library not found) )
+#AC_CHECK_LIB(stdc++,main,CXX_LIB="-lstdc++", AC_MSG_ERROR(stdc++ library not found) )
 AC_SUBST(CXX_LIB)
 
 AC_SUBST(LDLIBS)
Index: amsn
===================================================================
--- amsn	(revision 10906)
+++ amsn	(working copy)
@@ -1,6 +1,5 @@
-#!/bin/sh
-# \
-exec wish $0 $@
+#!/usr/bin/wish
+# 
 
 ###
 ###
Index: utils/linux/capture/capture.c
===================================================================
--- utils/linux/capture/capture.c	(revision 10906)
+++ utils/linux/capture/capture.c	(working copy)
@@ -1032,6 +1032,8 @@
     ng_debug = 0;
 # endif
   ng_init();
+
+  yuv2rgb_init();
   
   // End of Initialisation
   return TCL_OK;
